### DISCUSSION TITLE
Giving unique names to tiles in a TileMap [Solved]

### DISCUSSION THREAD

**Discussion by valkyrienyanko** [godotengine/godot-proposals#5316] [2D]
I'm developing a 2D platformer and I'm using TileMaps to create the world. If I could give unique names to tiles, I could do the following in script.

```cs
if (tile.Name == "water")
  // slow down player

if (tile.Name == "ice")
  // decrease player friction

if (tile.Name == "lava")
  // kill player
```

As suppose to

```cs
// creating comments to keep track of what is what
// 0 = air
// 6 = water
// 23 = ice
// 2 = lava

if (tile.Id == 6)
  // slow down player

if (tile.Id == 23)
  // decrease player friction

if (tile.Id == 2)
  // kill player
```

This would make scripts a whole lot less confusing to work with.

Note that `TileSet.TileGetName(tileId)` does not help me as it only returns the name of the file followed by the id of the tile (e.g. `tiles_packed.png 0` `tiles_packed.png 1`)

**Comment by timothyqiu** [comment]
You can currently use `TileSet`'s `find_tile_by_name()` in order to avoid magic numbers.

**Reply by valkyrienyanko** [reply]
I remember playing around with the tilemaps name functions and they returned the file name of the tileset followed by the id of the tile (`tiles_packed.png 1` `tiles_packed.png 2`) This does not solve my issue. What is the point of finding a tile by its name if you can't assign unique names to tiles in the first place? I want to be able to assign unique names to tiles and refer to these names later in script.

This is the code I used to get the tile names.

```cs
public override void _Process(float delta)
{
    var tileTerrainName = GetCurrentTileName(_tileMapTerrain, Position);
    GD.Print(tileTerrainName);
}

private string GetCurrentTileName(TileMap tilemap, Vector2 pos)
{
    var cellPos = tilemap.WorldToMap(pos);
    var cellId = tilemap.GetCellv(cellPos);

    if (cellId != TileMap.InvalidCell)
        return tilemap.TileSet.TileGetName(cellId);

    return null;
}
```

![image](https://user-images.githubusercontent.com/6277739/188325007-e4f4375c-0be9-4a4b-98a0-441236c64e08.png)

**Reply by TamerSoup625** [reply]
You actually _can_ assign an unique name to a tile.
In 3.x, you have to open the TileSet editor, select the tile, and then edit the "name" property in the inspector.

![(_) mess2 tscn - Messin with Godot 3 - Godot Engine 04_09_2022 20_38_21 (2)](https://user-images.githubusercontent.com/107559799/188328723-d1d56a87-6911-4d98-9f6f-4d6f8851c149.png)

**Reply by valkyrienyanko** [reply]
You are the best. Why have I not see this before!!!!!!!!!

**Reply by valkyrienyanko** [reply]
Can I do this in Godot 4 tileset as well? I can't find any option for it. (nvm figured it out)
